### PR TITLE
User Signup

### PR DESCRIPTION
**Pull request by emilynairnn**
- Created registration page for first time users
- Adds user information to the database
- Created authentication to ensure user is signing up properly
- POST request to database using sequelize

### CODE REVIEW

**Review by lkdfount — APPROVED** [review]
This looks awesome thank you both for doing it!!

**Review by lucasungaretti — COMMENTED** [review]
This looks pretty good. Im going to do some unit tests to make sure everything is going well.

File: travvy-server/app.js [hunk 1]
```diff
@@ -27,3 +27,11 @@ sequelize.sync().then(() => {
   app.listen(process.env.PORT || 5000)
   console.log(`Server sarted on port ${config.port}`)
 })
+// Send user signup information to database
+app.post('/signup',(req,res) =>{
```
Comment (dyampols) [inline travvy-server/app.js:31]: What is the purpose of this in the app.js file?

File: travvy-server/routes/routes.js [hunk 2]
```diff
@@ -26,4 +29,13 @@ module.exports = (app) => {
       .then(users => res.status(200).send(users))
       .catch(error => res.status(500).send(error))
     })
+ 
+ app.post('/signup', (req, res) => {
```
Comment (dyampols) [inline travvy-server/routes/routes.js:33]: will have to add logic to make sure that the user is not already registered

File: travvy-server/src/controllers/AuthenticationController.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
+
+
+const Users = require('../models').Users
+
+// This is the logic for adding users to the database
+// The format is set to match the Users table on MySQL
+// The null values will be replaced with information that the user will provide during onboarding
+module.exports = {
+  async signup (email,name, password) {
+    const user = await Users.create({
```
Comment (dyampols) [inline travvy-server/src/controllers/AuthenticationController.js:10]: Changed to  "users" for now to reflect the database even though it is one user. Need to keep in mind to stay consistent in the future

**Review by dyampols — COMMENTED** [review]
LGTM - if I could get an explanation to the app.js file

**Comment by lucasungaretti** [comment]
For my unit tests:
tested name,email,password combinations, noted potential issues to fix in future sprints:
1. all "Lucas" - added to database - ensure valid email 
2. email: lucas@gmail. com - works
3. second account with same email - did not register, good
4. no password - still registered, need to ensure all are filled
5. no name - same as above
6. no email - did not register, good
7. all "a" - no min requirement, maybe pw should be at least X number of digits

These are all things we can discuss when sprint planning and potentially include in next sprint.

**Comment by emilynairnn** [comment]
> For my unit tests:
> tested name,email,password combinations, noted potential issues to fix in future sprints:
> 
> 1. all "Lucas" - added to database - ensure valid email
> 2. email: lucas@gmail. com - works
> 3. second account with same email - did not register, good
> 4. no password - still registered, need to ensure all are filled
> 5. no name - same as above
> 6. no email - did not register, good
> 7. all "a" - no min requirement, maybe pw should be at least X number of digits
> 
> These are all things we can discuss when sprint planning and potentially include in next sprint.

I totally agree - there is a lot more error handling and testing that can be done in the next sprint.

**Pull request merged**